project/dexter-vs/Dexter/Analyzer/Tests/NoCommentAnalyzerTest.cs

```diff
@@ -0,0 +1,768 @@
+﻿using NUnit.Framework;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System;
+using Dexter.Analyzer.Tests.Helpers;
+using Dexter.Analyzer;
+
+namespace Dexter.Analyzer.Tests
+{
+    [TestFixture]
+    public class NoCommentAnalyzerTest : CodeFixVerifier
+    {
+        [Test]
+        public void NoCommentAnalyzer_verfiy_GivenClassTypeWithoutDoxygenComment()
+        {
+            var test = @"
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using System.Diagnostics;
+
+    namespace ConsoleApplication1
+    {
+        public class TestClass
+        {   
+        }
+    }";
+            var expected = new DiagnosticResult
+            {
+                Id = NoCommentAnalyzer.NoCommentRuleId,
+                Message = "API Type 'TestClass' has no doxygen comment",
+                Severity = DiagnosticSeverity.Warning,
+                Locations =
+                    new[] {
+                            new DiagnosticResultLocation("Test0.cs", 11, 22)
+                        }
+            };
+
+            VerifyCSharpDiagnostic(test, expected);
+        }
+
+        [Test]
+        public void NoCommentAnalyzer_Ignore_GivenNotPublicClassTypeWithoutDoxygenComment()
+        {
+            var test = @"
```

Comment:
It would be better to keep the current form because it is easy to add test cases and see what we are testing.